subsys/logging/log_core.c

```diff
@@ -395,13 +395,27 @@ void log_generic(struct log_msg_ids src_level, const char *fmt, va_list ap)
 		}
 	} else {
 		log_arg_t args[LOG_MAX_NARGS];
-		u32_t nargs = count_args(fmt);
+		u32_t nargs = log_count_args(fmt);
 
 		__ASSERT_NO_MSG(nargs < LOG_MAX_NARGS);
 		for (int i = 0; i < nargs; i++) {
 			args[i] = va_arg(ap, log_arg_t);
 		}
 
+		if (enable_strdup) {
+			u32_t mask = z_log_get_s_mask(fmt, nargs);
+
+			while (mask) {
+				u32_t idx = 31 - __builtin_clz(mask);
+				const char *str = (const char *)args[idx];
+
+				if (!log_is_strdup(str) &&
+					(str != log_strdup_fail_msg)) {
```

Comment:
str != log_strdup_fail_msg this test is most probably not needed since log_strdup tests for is_rodata and log_strdup_fail_msg is in ro area.